addons/binding/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/handler/ChromecastHandler.java

```diff
@@ -296,82 +368,199 @@ private void handleMute(final Command command) {
                 updateStatus(ThingStatus.ONLINE);
             } catch (final IOException ex) {
                 logger.debug("Mute/unmute volume failed: {}", ex.getMessage());
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getMessage());
+                updateStatus(ThingStatus.OFFLINE, COMMUNICATION_ERROR, ex.getMessage());
             }
         }
     }
 
-    private void scheduleStopApp() {
-        scheduler.schedule(() -> {
-            try {
-                Application app = chromecast.getRunningApp();
-                if (app.id.equals(MEDIA_PLAYER) && this.appSessionId != null
-                        && app.sessionId.equals(this.appSessionId)) {
-                    chromecast.stopApp();
-                    logger.debug("Media player app stopped");
-                }
-            } catch (final Exception e) {
-                logger.debug("Failed stopping media player app: {}", e.getMessage());
-            }
-        }, 0, TimeUnit.MILLISECONDS);
+    @Override
+    public void connectionEventReceived(ChromeCastConnectionEvent event) {
+        if (event.isConnected()) {
+            updateStatus(ThingStatus.ONLINE);
+            scheduleRefresh();
+        } else {
+            cancelRefresh();
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    }
+
+    @Override
+    public void spontaneousEventReceived(final ChromeCastSpontaneousEvent event) {
+        switch (event.getType()) {
+            case MEDIA_STATUS:
+                updateMediaStatus(event.getData(MediaStatus.class));
+                break;
+            case STATUS:
+                updateChromecastStatus(event.getData(Status.class));
+                break;
+            case UNKNOWN:
+                logger.debug("Received an 'UNKNOWN' event (class={})", event.getType().getDataClass());
+                break;
+            default:
+                logger.debug("Unhandled event type: {}", event.getData());
+                break;
+        }
     }
 
-    private void handleCcStatus(final Status status) {
+    @Override
+    protected void updateStatus(ThingStatus status, ThingStatusDetail statusDetail, String description) {
+        if (status == ThingStatus.ONLINE) {
+            updateState(new ChannelUID(getThing().getUID(), CHANNEL_ONLINE_STATUS), OnOffType.ON);
```

Comment:
Could you change all `updateState` calls to simply be like this:
```
updateState(CHANNEL_ONLINE_STATUS, OnOffType.ON);
```
No need to instantiate a ChannelUID.